app/src/app/event/health-check-ping-executor.ts

```diff
@@ -36,23 +36,20 @@ export class HealthCheckPingExecutor {
 
     private async pingError(_err: Error): Promise<void> {
         try {
-            await Resources.network().post(this.healthCheckUrl + `/fail`, this.getLog());
+            await this.getLog().then(log => this.networkInterface.post(`/fail`, log));
             Resources.logger(this).info(`Successfully sent error health check ping.`);
         } catch (e) {
             Resources.logger(this).error(`Failed to send error health check ping: ${e}`);
         }
     }
 
-    private getLog(): string {
-        // Get roughly the 100KB by getting the last 100 characters
-        // may be too large due to some characters taking up more than one byte
-        const last100Chars = this
-            .logInterface
-            .getLog()
-            .slice(-100000);
-
-        // Actually get the last 100KB by encoding the last 100 characters
-        const bytes = new TextEncoder().encode(last100Chars);
-        return new TextDecoder().decode(bytes.slice(-100000));
+    private async getLog(): Promise<string> {
+        const filePath = Resources.manager().logFilePath;
+        if (!filePath) {
+            return `no log available`;
+        }
+
+        return fs.readFile(filePath)
+            .then(data => data.subarray(-1000000).toString(FILE_ENCODING))
```

Comment:
Definitely nicer implementation of getting the last 1000 bytes. Question is what happens if we cut within a character? E.g. we only take the last 3 bytes of the 4 bytes that make up "😃". This might make the log unreadable for healthchecks.io , but this should be easy to try out. I think my implementation had the same problem.

Reply:
Commented on an old version of the file but the question still applied. I might have some time to try this out tomorrow or the day after.